src/main/java/com/rabbitmq/client/ConnectionFactoryConfigurer.java

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) 2017-Present Pivotal Software, Inc.  All rights reserved.
+//
+// This software, the RabbitMQ Java client library, is triple-licensed under the
+// Mozilla Public License 1.1 ("MPL"), the GNU General Public License version 2
+// ("GPL") and the Apache License version 2 ("ASL"). For the MPL, please see
+// LICENSE-MPL-RabbitMQ. For the GPL, please see LICENSE-GPL2.  For the ASL,
+// please see LICENSE-APACHE2.
+//
+// This software is distributed on an "AS IS" basis, WITHOUT WARRANTY OF ANY KIND,
+// either express or implied. See the LICENSE file for specific language governing
+// rights and limitations of this software.
+//
+// If you have any questions regarding licensing, please contact us at
+// info@rabbitmq.com.
+
+package com.rabbitmq.client;
+
+import com.rabbitmq.client.impl.AMQConnection;
+import com.rabbitmq.client.impl.nio.NioParams;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Reader;
+import java.net.URISyntaxException;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * Helper class to load {@link ConnectionFactory} settings from a property file.
+ *
+ * The authorised keys are the constants values in this class (e.g. USERNAME).
+ * The property file/properties instance/map instance keys can have
+ * a prefix, the default being <code>rabbitmq.</code>.
+ *
+ * Property files can be loaded from the file system (the default),
+ * but also from the classpath, by using the <code>classpath:</code> prefix
+ * in the location.
+ *
+ * If default client properties should be set, set the <code>use.default.client.properties</code>
+ * key to <code>true</code>. Custom client properties can be set by using
+ * the <code>client.properties.</code>, e.g. <code>client.properties.app.name</code>.
+ *
+ * @since 4.4.0
+ * @see ConnectionFactory#load(String, String)
+ */
+public class ConnectionFactoryConfigurer {
+
+    public static final String DEFAULT_PREFIX = "rabbitmq.";
+
+    public static final String USERNAME = "username";
+    public static final String PASSWORD = "password";
+    public static final String VIRTUAL_HOST = "virtual.host";
+    public static final String HOST = "host";
+    public static final String PORT = "port";
+    public static final String REQUESTED_CHANNEL_MAX = "requested.channel.max";
+    public static final String REQUESTED_FRAME_MAX = "requested.frame.max";
+    public static final String REQUESTED_HEARTBEAT = "requested.heartbeat";
```

Comment:
I'd use "heartbeat.timeout" here. Same for `frame.max` and `channel.max`.

Reply:
Again, I tried to map with the existing properties in `ConnectionFactory`. Consistency over better naming?  :-)

Reply:
It's just weird that we use dots so aggressively. In case of `requested.heartbeat` we propagate an unfortunately named property and make it even more confusing by using dot separators.

Can we use `connection.channel_max`, `connection.frame_max`, `connection.heartbeat.timeout` for those 3?

Reply:
Sure.